### PR TITLE
introduces the gRPC API for Authzed Materialize

### PR DESCRIPTION
**Pull request by vroldanbet**
This is published as v0 and is considered alpha version

### CODE REVIEW

File: authzed/api/v0/materialize/watchpermissions.proto [hunk 1]
```diff
@@ -0,0 +1,87 @@
+syntax = "proto3";
+package authzed.api.materialize.v0;
```
Comment (jzelinskie) [inline authzed/api/v0/materialize/watchpermissions.proto:2]: this package seems right, but the path for this file seems to be `authzed/api/v0/materialize` and not `authzed/api/materialize/v0`

**Review by jzelinskie — CHANGES_REQUESTED** [review]

File: authzed/api/v0/materialize/watchpermissions.proto [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  // WatchPermissions is a long-running RPC, and will stream events until the client
+  // closes the connection or the server terminates the stream. The consumer is responsible of
+  // keeping track of the last seen revision and resuming the stream from that point in the event
+  // of disconnection or client-side restarts.
+  //
+  // The API does not offer a sharding mechanism and thus there should only be one consumer per target system.
+  // Implementing an active-active HA consumer setup over the same target system will require coordinating which
+  // revisions have been consumed in order to prevent transitioning to an inconsistent state.
+  //
+  // Usage of WatchPermissions requires it to be explicitly enabled on the service,
```
Comment (josephschorr) [inline authzed/api/v0/materialize/watchpermissions.proto:24]: Indicate that the permissions watched needs to be explicitly set on the service side

File: authzed/api/v0/materialize/watchpermissions.proto [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  // requires more resources and is less performant than WatchPermissionsSets. It's usage
+  // is only recommended when performing the set intersections of WatchPermissionSets in the client side is not viable
+  // or there is a strict application requirement to use consume the computed permissions.
+  rpc WatchPermissions(WatchPermissionsRequest) returns (stream WatchPermissionsResponse) {}
+}
+
+message WatchPermissionsRequest {
+  // permissions is a list of permissions to watch for changes. At least one permission must be specified
```
Comment (josephschorr) [inline authzed/api/v0/materialize/watchpermissions.proto:32]: Indicate must be a subset of those enabled on the service

File: authzed/api/v0/materialize/watchpermissionsets.proto [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  //
+  // In exchange, the number of changes issued by WatchPermissionSets will be several orders of magnitude less than those
+  // emitted by WatchPermissions, which has several implications:
+  // - significantly less resources to compute the sets
+  // - significantly less messages to stream over the network
+  // - significantly less events to ingest on the consumer side
+  // - less ingestion lag from the origin SpiceDB mutation
+  //
+  // The type of scenarios WatchPermissionSets is particularly well suited is when a single change
+  // in the origin SpiceDB can yield millions of changes. For example, in the GitHub authorization model, assigning a role
```
Comment (josephschorr) [inline authzed/api/v0/materialize/watchpermissionsets.proto:35]: link the sample schema

**Review by jzelinskie — APPROVED** [review]

**Review by josephschorr — APPROVED** [review]

**Pull request merged**